### PR TITLE
How to use GitHub’s search feature to search the wiki

### PR DESCRIPTION
**Pull request by MohitGupta121**
## Explanation

How to use GitHub’s search feature to search the wiki
- With this contributor easily search and navigate to desire wiki page.
- Added TOC in all wikis pages.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Review by MohitGupta121 — COMMENTED** [review]
@BenHenning @seanlip PTAL,

In last welfare meeting I discussed the problem that we have enough infomation in wiki but contributors not able to reach it. So the solution is GitHub search feature which helps to search into wiki (I think most of the new contributors even I not know that apart from code, we also search into wiki with GitHub search feature.).

So I decied to add Gif to demostrate this feature both in oppia and oppia-android after discussing with Chris.
Also I decided to add missing TOC in oppia-android wiki page which make searcing easy.

Thanks.

**Review by seanlip — CHANGES_REQUESTED** [review]
Thanks @MohitGupta121. Two questions:

(a) Perhaps this should be on the "Coders" page instead, since the home page is meant to be more generic for different types of volunteers? Alternatively, you can have it at the bottom of the page in a separate section ("How to find information on this wiki").

(b) You mention adding a TOC, but I don't see it?

**Comment by MohitGupta121** [comment]
@seanlip Thanks for review,
a) Actually Chris suggest to keep it on Home page that's why, But I will move it to coders.
b) Regarding TOC, I waiting for your suggestions (if any) to go ahead, that's why I not update it here. But now you allow I will update that also.

Thanks.

**Comment by seanlip** [comment]
Hm, for (b), this wasn't clear from your previous comment. I can't approve it since I don't know what you are proposing exactly, but in the future I suggest making the changes you think would be best and then sending that entire PR for review -- it speeds up the process.

**Comment by MohitGupta121** [comment]
Okay Thanks @seanlip, I keep this in mind and act accordingly.

**Review by MohitGupta121 — COMMENTED** [review]
@seanlip PTAL, Thanks.
I added TOC to most of the wiki related to oppia-android codebase. But not for some general wikis such as "Our Mission", and "SLoP-2020".
If we need TOC for that wikis also please let me know.
Thanks.

File: wiki/Overview-of-the-Oppia-Android-codebase-and-architecture.md
Comment (seanlip) [inline wiki/Overview-of-the-Oppia-Android-codebase-and-architecture.md]: @MohitGupta121 this should use spaces not tabs, I think, for consistency with all the others.

**Review by seanlip — CHANGES_REQUESTED** [review]
Thanks @MohitGupta121, I generally have no concerns. Just noticed one minor thing.

File: wiki/Overview-of-the-Oppia-Android-codebase-and-architecture.md
Comment (MohitGupta121) [reply wiki/Overview-of-the-Oppia-Android-codebase-and-architecture.md]: Okay @seanlip, for all TOC, I have to use space?

Comment (seanlip) [reply wiki/Overview-of-the-Oppia-Android-codebase-and-architecture.md]: Yeah, I think you do so everywhere else though? Just for consistency.

**Review by seanlip — APPROVED** [review]
Thanks @MohitGupta121, LGTM!

**Comment by MohitGupta121** [comment]
@seanlip PTAL, replace all other Tabs to spaces.
Thanks.

File: wiki/Overview-of-the-Oppia-Android-codebase-and-architecture.md
Comment (MohitGupta121) [reply wiki/Overview-of-the-Oppia-Android-codebase-and-architecture.md]: Done.

**Review by BenHenning — APPROVED** [review]
Thanks @MohitGupta121! LGTM for textproto change.

Note I didn't review the wiki files in detail since @seanlip already approved, but I can take a pass if you need me to.

File: wiki/Contributing-to-Oppia-Android.md
Comment (MohitGupta121) [inline wiki/Contributing-to-Oppia-Android.md:19]: @BenHenning Thanks for review, Actually I only added this demo for wiki search in github. You can review this.
Apart from this I added only TOC in all wiki pages.

Comment (BenHenning) [reply wiki/Contributing-to-Oppia-Android.md:19]: Thanks! This seems reasonable. We may need to change it in a few months when GitHub's new search functionality exits beta, but it should work in the interim. :)

**Review by BenHenning — APPROVED** [review]
Thanks @MohitGupta121! The new search feature LGTM as well.

File: wiki/Contributing-to-Oppia-Android.md
Comment (MohitGupta121) [reply wiki/Contributing-to-Oppia-Android.md:19]: Yes sure, I will change that.

**Comment by MohitGupta121** [comment]
@BenHenning Do I need to add new TODO issue for wiki update when GitHub new search feature release?

And Will you please merge this, if nothing unresolved. Thanks!

**Comment by BenHenning** [comment]
> @BenHenning Do I need to add new TODO issue for wiki update when GitHub new search feature release?
> 
> And Will you please merge this, if nothing unresolved. Thanks!

A TODO isn't necessary, I think. We can update this reactively as the flow on GitHub isn't changing too drastically.

**Pull request merged**
